### PR TITLE
Solved bug with incompatible QoS policies

### PR DESCRIPTION
**Pull request by tudoroancea**
### Public-Facing Changes

None

### Description

I changed 3 files for 3 different reasons:
1. **`CMakeLists.txt`**: fixed build issue with websocket++ that wasn't found properly. The issue was also mentioned in [this other PR](https://github.com/foxglove/ros-foxglove-bridge/pull/230). In comparison with it, I did not add the `/usr/local/include` that wasn't cross-platform.
2. **`ros2_foxglove_bridge/src/parameter_interface.cpp`**: line 97 raised an error when compiling on macOS. My guess is that the default clang compilation flags don't allow implicit calls to constructors.
3. **`ros2_foxglove_bridge/src/ros2_foxglove_bridge.cpp`**: this solves the [following issue](https://github.com/tudoroancea/ros-foxglove-bridge/issues/1).

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -29,6 +29,8 @@ find_package(OpenSSL REQUIRED)
 find_package(Threads REQUIRED)
 find_package(websocketpp REQUIRED)
 find_package(ZLIB REQUIRED)
+find_path(WEBSOCKETPP_INCLUDE_DIR NAMES websocketpp/websocketpp.hpp)
+message(STATUS "WEBSOCKETPP_INCLUDE_DIR: " ${WEBSOCKETPP_INCLUDE_DIR})
```
[the remark this answers is not in the document]
Comment (tudoroancea) [reply CMakeLists.txt:33]: For some reason, when building `ros-foxglove-bridge` with a ROS 2 installation from robostack (in a conda env), CMake/clang installed in the same env, and with `websocketpp` installed globally via homebrew or apt, this variable was not populated correctly. However, if I install websocketpp in the conda env, it works just fine.
I will try to understand further the bug

File: ros2_foxglove_bridge/src/ros2_foxglove_bridge.cpp [hunk 2]
```diff
@@ -494,32 +494,41 @@ void FoxgloveBridge::subscribe(foxglove::ChannelId channelId, ConnectionHandle c
...
+      qos.transient_local();
+    } else {
+      if (durabilityTransientLocalEndpointsCount > 0) {
+        RCLCPP_WARN(this->get_logger(),
+                    "Some, but not all, publishers on topic '%s' are offering "
+                    "QoSDurabilityPolicy.TRANSIENT_LOCAL. Falling back to "
+                    "QoSDurabilityPolicy.VOLATILE as it will connect to all publishers",
+                    topic.c_str());
+      }
+      qos.durability_volatile();
```
[the remark this answers is not in the document]
Comment (tudoroancea) [reply ros2_foxglove_bridge/src/ros2_foxglove_bridge.cpp:530]: Yes, you are right. I just added a separate if condition to call `RCLCPP_WARN`. I will remove it in the next PR

**Comment by tudoroancea** [comment]
I am not sure why the CI failed tho. It did so with the first other PR I created: #277

File: CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (achim-k) [reply CMakeLists.txt:33]: You could also look into adapting the robotstack [foxglove_bridge patch](https://github.com/RoboStack/ros-humble/blob/main/patch/ros-humble-foxglove-bridge.patch). That patch might need some updating as it's relatively old, but building on MacOS worked back then

**Comment by achim-k** [comment]
@tudoroancea are you still working on this?

**Comment by achim-k** [comment]
@tudoroancea since both mentioned PRs are in, this can be closed now?

**Comment by mikaelarguedas** [comment]
> Build failures are related to https://github.com/osrf/docker_images/issues/697. Might take a bit until the base docker images are rebuild to include the updated key. If this doesn't happen, we will have to do that manually in our ROS1/ROS2 Dockerfiles.

FYI the ROS images on dockerhub have been rebuilt with the updated GPG key.
More details at https://github.com/osrf/docker_images/issues/697#issuecomment-1856463789

**Pull request closed without merging**